src/libs/actions/DemoActions.js

```diff
@@ -0,0 +1,92 @@
+import Onyx from 'react-native-onyx';
+import _ from 'underscore';
+import lodashGet from 'lodash/get';
+import CONST from '../../CONST';
+import * as API from '../API';
+import * as ReportUtils from '../ReportUtils';
+import Navigation from '../Navigation/Navigation';
+import ROUTES from '../../ROUTES';
+import ONYXKEYS from '../../ONYXKEYS';
+import * as Localize from '../Localize';
+
+/**
+ * @param {String} workspaceOwnerEmail email of the workspace owner
+ * @param {String} apiCommand
+ */
+function createDemoWorkspaceAndNavigate(workspaceOwnerEmail, apiCommand) {
+    // Try to navigate to existing demo workspace expense chat if it exists in Onyx
+    const demoWorkspaceChatReportID = ReportUtils.getPolicyExpenseChatReportIDByOwner(workspaceOwnerEmail);
+    if (demoWorkspaceChatReportID) {
+        // We must call goBack() to remove the demo route from nav history
+        Navigation.goBack();
+        Navigation.navigate(ROUTES.getReportRoute(demoWorkspaceChatReportID));
+        return;
+    }
+
+    // We use makeRequestWithSideEffects here because we need to get the workspace chat report ID to navigate to it after it's created
+    // eslint-disable-next-line rulesdir/no-api-side-effects-method
+    API.makeRequestWithSideEffects(apiCommand).then((response) => {
+        // Get report updates from Onyx response data
+        const reportUpdate = _.find(response.onyxData, ({key}) => key === ONYXKEYS.COLLECTION.REPORT);
```

Comment:
Did you mean to check if the key starts with `ONYXKEYS.COLLECTION.REPORT`? And are we guaranteed to always have one report update?

Reply:
We're receiving an onyx update here which has the `mergecollection` onyx method. So the key check is correct here.